pkg/frontend/admin_openshiftcluster_kubernetespods_logs_test.go

```diff
@@ -0,0 +1,194 @@
+	mockTenantID := "00000000-0000-0000-0000-000000000000"
+	ctx := context.Background()
+
+	type test struct {
+		name           string
+		resourceID     string
+		namespace      string
+		podName        string
+		containerName  string
+		mocks          func(*test, *mock_adminactions.MockKubeActions)
+		method         string
+		wantStatusCode int
+		wantResponse   []byte
+		wantError      string
+	}
+
+	for _, tt := range []*test{
+		{
+			method:        http.MethodGet,
+			name:          "cluster exist in db",
+			resourceID:    fmt.Sprintf("/subscriptions/%s/resourcegroups/resourceGroup/providers/Microsoft.RedHatOpenShift/openShiftClusters/resourceName", mockSubID),
+			namespace:     "openshift-project",
+			podName:       "config",
+			containerName: "container-1",
+			mocks: func(tt *test, k *mock_adminactions.MockKubeActions) {
+				k.EXPECT().
+					KubeGetPodLogs(gomock.Any(), tt.namespace, tt.podName, tt.containerName).
+					Return([]byte(`{"Kind": "test"}`), nil)
+
+			},
+			wantStatusCode: http.StatusOK,
+			wantResponse:   []byte(`{"Kind": "test"}` + "\n"),
+		},
+	} {
+		t.Run(fmt.Sprintf("%s: %s", tt.method, tt.name), func(t *testing.T) {
+			ti := newTestInfra(t).WithOpenShiftClusters().WithSubscriptions()
+			defer ti.done()
+			k := mock_adminactions.NewMockKubeActions(ti.controller)
+
+			tt.mocks(tt, k)
+
+			ti.fixture.AddOpenShiftClusterDocuments(&api.OpenShiftClusterDocument{
+				Key: strings.ToLower(tt.resourceID),
+				OpenShiftCluster: &api.OpenShiftCluster{
+					ID:   tt.resourceID,
+					Name: "resourceName",
+					Type: "Microsoft.RedHatOpenShift/openshiftClusters",
+					Properties: api.OpenShiftClusterProperties{
+						NetworkProfile: api.NetworkProfile{
+							APIServerPrivateEndpointIP: "0.0.0.0",
+						},
+					},
+				},
+			})
+			ti.fixture.AddSubscriptionDocuments(&api.SubscriptionDocument{
+				ID: mockSubID,
+				Subscription: &api.Subscription{
+					State: api.SubscriptionStateRegistered,
+					Properties: &api.SubscriptionProperties{
+						TenantID: mockTenantID,
+					},
+				},
+			})
+
+			err := ti.buildFixtures(nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			f, err := NewFrontend(ctx, ti.audit, ti.log, ti.env, ti.asyncOperationsDatabase, ti.openShiftClustersDatabase, ti.subscriptionsDatabase, api.APIs, &noop.Noop{}, nil, func(*logrus.Entry, env.Interface, *api.OpenShiftCluster) (adminactions.KubeActions, error) {
+				return k, nil
+			}, nil, nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			go f.Run(ctx, nil, nil)
+
+			resp, b, err := ti.request(tt.method,
+				fmt.Sprintf("https://server/admin%s/kubernetespodlogs?container=%s&namespace=%s&name=%s", tt.resourceID, tt.containerName, tt.namespace, tt.podName),
+				nil, nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			err = validateResponse(resp, b, tt.wantStatusCode, tt.wantError, tt.wantResponse)
+			if err != nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
+func TestValidateAdminKubernetesPodLogs(t *testing.T) {
```

Comment:
IMO, this test should be in `pkg/frontend/validate_test.go` becasue the function we test is in `pkg/frontend/validate.go`.

The test itself looks good to me.

Reply:
There are already validations present for other admin actions in validate.go. It looks like every admin action has its own validation that is why the tests for the same was kept separately. I too think that the tests for validations for different actions should be isolated and hence should be in respective action's test file.

Reply:
@anshulvermapatel we have to start somewhere :) We can start by moving this test into `pkg/frontend/validate_test.go` and then in a separate PR we can update other tests.

Reply:
Okkay .. I will shift the tests to `pkg/frontend/validate_test.go`